### ISSUE TITLE
Automatically change ContainerProbeSettings to true when using custom probes

### ISSUE THREAD

**Issue by junior-juarez-MSFT** [Azure/Azure-Spring-Apps#57]
**Is your feature request related to a problem? Please describe.**

Yes, this feature request is related to a report that is already closed https://github.com/Azure/Azure-Spring-Apps/issues/15 and support ticket.

**Describe the solution you'd like**

It will be good when switching from default `ContainerProbeSettings.disableProbe = FALSE` to custom readiness and liveness probe settings that ASA automatically change `ContainerProbeSettings.disableProbe` to `TRUE`. This way customers can avoid the error message:

```
There was a problem saving the configuration. Custom Probe feature cannot be used with ContainerProbeSettings together..
```

**Describe alternatives you've considered**

This solution can be applied for IaC scripts like ARM/Bicep/Terraform and Azure CLI.

**Describe the Customer Impact**

Customers have to redeploy all their app deployments that are already using `ContainerProbeSettings` which is a tedious work that have to be done and may cause delay in their deadlines.

**Additional context**

**Can we contact you for additional details?** Y/N

If yes, please send us your contact information to AzureSpringCloud-Talk@service.microsoft.com and include the issue number in the email title.

**Comment by Caoxuyang** [comment]
Hey @junior-juarez-MSFT, thx for the suggestion. 

But I don't quite understand the issue. 
The background is that we have a scenario, the customers used to use "ContainerProbeSettings.disableProbe" and now they want to turn around to use the custom prove feature, right? 

No matter what deployment tools they are using, they have to edit all their apps with "ContainerProbeSettings.disableProbe", at least they need to add properties like liveness probes/readiness probes, is that right? 

So they just need to delete the "ContainerProbeSettings.disableProbe" property at the same time, no additional redeploy at all. They always have to do one redeploy when switching the feature.

**Comment by junior-juarez-MSFT** [comment]
@Caoxuyang 

Yes, you are right they have to redeploy everything again if custom probe settings are going to be set.

What we would like is that `ContainerProbeSettings.disableProbe` can be omitted during the Azure CLI or IaC deployment and be automatically changed to `TRUE` when custom settings are set and the other way around, as initially `ContainerProbeSettings.disableProbe` are going to be either `TRUE` or `FALSE` depending on what settings are originally used.

**Comment by Caoxuyang** [comment]
@junior-juarez-MSFT I was looped in the email thread. I will address the question in email thread and update the result later back here in github.
